### ISSUE TITLE
Workflow tests

### ISSUE THREAD

**Issue by tisaconundrum2** [USGS-Astrogeology/PySAT_Point_Spectra_GUI#86]
Testing workflows

Load Data Set 1
Load Data Set 2
Interpolate Data Set 2 onto Data Set 1
Mask Data Set 1 and 2
Normalize Data Set 1 and 2
Strat Folds Data Set 1
CV Data Set 1 - Train (PLS, nc=1-10)
Plot RMSECV vs # of components from the CV Results
Export CV Results to .csv
Train PLS on Data Set 1 - Train (nc=6)
Predict Data Set 1 - Test
Predict Data Set 2
Plot Data Set 1 - Test predictions on 1 to 1 plot
Plot Data Set 2 predictions on the same plot

Load data set (the JSC data is good for this since it is not already baseline corrected)
Mask
Baseline Removal (AirPLS)
Plot Spectra - Original data set - pick a spectrum based on target name
Plot Spectra - Baseline for the same spectrum on the same plot, but different color

Load Data Set
Mask Data Set
Normalize Data Set
Strat Folds Data Set
Train PLS on Data Set - Train (nc=6, 0-100%)
Train PLS on Data Set - Train (nc=7, 60-100%)
Train PLS on Data Set - Train (nc=5, 30-70%)
Train PLS on Data Set - Train (nc=4, 0-40%)
Train Lasso on Data Set - Train (use CV optimization)
Submodel predict Data set - Test using PLS 0-100% as reference, 0-40 as low, 30-70 as mid, 60-100 as high model
Predict Data Set - Test using Lasso
Plot both predictions on 1 to 1 plot

**Comment by tisaconundrum2** [comment]
## example_cv.wrf
 - Load one set of data that will be used to create the model
 - Load a second set of data that will serve as the "unknown" (even though in this test case I'm using some known data I had on hand)
 - Set output folder (this is where things like plots or modified data files will be saved. Note: if you save workflows, they actually go into the point_spectra_gui directory. I need to fix this...)
 - Interpolate the "unknown" data to the same wavelengths as the "known" data
 - Mask out unwanted portions the first data set (example mask files are now included in the "inputs" directory)
 - Apply the same mask to the second data set
 - Normalize the spectra in the first data set so that the sum of each spectrum (from 0-1000 nm) is 1
 - Same normalization on the second data set
 - Get rid of rows in the first data set that don't have compositions for SiO2 (if you don't do this, it causes problems later on...)
 - Divide the first data set into 5 folds with similar distributions of SiO2 compositions. Set fold 3 to be used as the test set, use the remaining folds as a training set.
 - Run cross validation for PLS from 1 to 15 components
 - Plot the cross validation results as a function of number of components (RMSECV = root mean squared error of cross validation. i.e. how does the model do when each fold is held out in turn and a model is trained with the remaining folds?)
 - Overplot the RMSEC results (i.e. how well does the training set predict itself? This tends to always be overly-optimistic)
 - Based on these plots, the error stops improving significantly around 8 components, so that's what should be used.

## example_train_predict.wrf
 - Same preprocessing steps through defining folds.
 - Train a PLS regression model for SiO2 using the optimum number of components that we found from cross validation (nc = 8)
 - Use that model to predict the test set composition
 - Plot the test set prediction against the actual composition to see how well the model did, save to the file "SiO2_lab_test.png"
 - Train a second model using the lasso algorithm (this algorithm does its own internal cross validation, so I just left it with default settings)
 - Predict the test set with the Lasso model
 - Plot the Lasso results on the same plot with different colored points, saved to the file "SiO2_lab_test.png"
